src/mito2/src/worker/handle_remap.rs

```diff
@@ -112,12 +114,34 @@ impl<S> RegionWorkerLoop<S> {
             .collect::<Result<HashMap<_, _>>>()?;
         let mut mapper = RemapManifest::new(manifests, new_partition_exprs, region_mapping);
         let remap_result = mapper.remap_manifests()?;
+
+        // Write new manifests to staging blob storage.
+        let manifest_manager = region.manifest_ctx.manifest_manager.write().await;
+        let manifest_storage = manifest_manager.store();
+        let staging_blob_storage = manifest_storage.staging_storage().blob_storage().clone();
+        let mut tasks = Vec::with_capacity(remap_result.new_manifests.len());
+
+        for (remap_region_id, manifest) in &remap_result.new_manifests {
+            let bytes = serde_json::to_vec(&manifest).context(error::SerializeManifestSnafu {
+                region_id: *remap_region_id,
+            })?;
+            let key = remap_region_id.as_u64().to_string();
+            tasks.push(async {
+                debug!(
+                    "Putting manifest to staging blob storage, region_id: {}, key: {}",
+                    *remap_region_id, key
+                );
+                staging_blob_storage.put(&key, bytes).await?;
```

Comment:
Do we need to add some other string to the key instead of only using a single region id?